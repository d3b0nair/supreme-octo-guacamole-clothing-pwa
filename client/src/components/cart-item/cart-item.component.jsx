import React from "react";

import {
  CartItemContainer,
  CartItemImg,
  ItemDetailsContainer,
} from "./cart-item.styles";

export function CartItem({ item: { imageUrl, price, name, quantity } }) {
  return (
    <CartItemContainer>
      <CartItemImg src={`${window.location.origin}/${imageUrl}`} alt={name} />
      <ItemDetailsContainer>
        <span>{name}</span>
        <span>
          {quantity} x ${price}
        </span>
      </ItemDetailsContainer>
    </CartItemContainer>
  );
}
export default React.memo(CartItem);
