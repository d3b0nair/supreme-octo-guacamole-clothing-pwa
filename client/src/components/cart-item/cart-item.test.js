import React from "react";
import { render, cleanup } from "@testing-library/react";
import { CartItem } from "./cart-item.component";

const mockProps = (otherProps) => ({
  ...otherProps,
  item: {
    imageUrl: "",
    price: 1,
    name: "",
    quantity: 1,
  },
});
afterEach(cleanup);

const component = (mockProps) => <CartItem {...mockProps} />;
describe("CartIcon component", () => {
  it("should mount", () => {
    const { container } = render(component(mockProps()));
    expect(container).toMatchSnapshot();
  });
});
