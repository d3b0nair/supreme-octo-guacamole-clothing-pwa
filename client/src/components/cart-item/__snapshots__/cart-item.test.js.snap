// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartIcon component should mount 1`] = `
<div>
  <div
    class="sc-bdvvtL bpFOcX"
  >
    <img
      alt=""
      class="sc-gsDKAQ jCtDRh"
      src="http://localhost/"
    />
    <div
      class="sc-dkPtRN beJrup"
    >
      <span />
      <span>
        1
         x $
        1
      </span>
    </div>
  </div>
</div>
`;
